src/operator.rs

```diff
@@ -600,7 +612,16 @@ impl UnaryPreAggFunction for Parse {
     fn process(&self, rec: Record) -> Result<Option<Record>, EvalError> {
         let matches = self.matches(&rec)?;
         match matches {
-            None => Ok(None),
+            None => match self.options.drop_nonmatching {
```

Comment:
I think the rust style is to use an if-expression instead of matching on a boolean.

Reply:
I'd probably use tuple matching and do something like
```
match (self.options,drop_nonmatching, matches) {
  None, true => None,
  None, false => addFields(rec)
  Some(rec), _ => rec
}
```